Content.Client/UserInterface/Controls/ListContainer.cs

```diff
@@ -0,0 +1,359 @@
+        }
+        finally
+        {
+            _suppressScrollValueChanged = false;
+        }
+
+        if (_vScrollBar.Visible)
+        {
+            _vScrollBar.Arrange(UIBox2.FromDimensions(Vector2.Zero, finalSize));
+        }
+        #endregion
+
+        #region Rebuild Children
+        /*
+         * Example:
+         *
+         * var _itemHeight = 32;
+         * var separation = 3;
+         *  32 | 32 | Control.Size.Y 0
+         *  35 |  3 | Padding
+         *  67 | 32 | Control.Size.Y 1
+         *  70 |  3 | Padding
+         * 102 | 32 | Control.Size.Y 2
+         * 105 |  3 | Padding
+         * 137 | 32 | Control.Size.Y 3
+         *
+         * If viewport height is 60
+         * visible should be 2 items (start = 0, end = 1)
+         *
+         * scroll.Y = 11
+         * visible should be 3 items (start = 0, end = 2)
+         *
+         * start expected: 11 (item: 0)
+         * var start = (int) (scroll.Y
+         *
+         * if (scroll == 32) then { start = 1 }
+         * var start = (int) (scroll.Y + separation / (_itemHeight + separation));
+         * var start = (int) (32 + 3 / (32 + 3));
+         * var start = (int) (35 / 35);
+         * var start = (int) (1);
+         *
+         * scroll = 0, height = 36
+         * if (scroll + height == 36) then { end = 2 }
+         * var end = (int) Math.Ceiling(scroll.Y + height / (_itemHeight + separation));
+         * var end = (int) Math.Ceiling(0 + 36 / (32 + 3));
+         * var end = (int) Math.Ceiling(36 / 35);
+         * var end = (int) Math.Ceiling(1.02857);
+         * var end = (int) 2;
+         *
+         */
+        var scroll = GetScrollValue();
+        var oldTopIndex = _topIndex;
+        _topIndex = (int) ((scroll.Y + ActualSeparation) / (_itemHeight + ActualSeparation));
+        if (_topIndex != oldTopIndex)
+            _updateChildren = true;
+
+        var oldBottomIndex = _bottomIndex;
+        _bottomIndex = (int) Math.Ceiling((scroll.Y + finalHeight) / (_itemHeight + ActualSeparation));
+        _bottomIndex = Math.Min(_bottomIndex, _data.Count);
+        if (_bottomIndex != oldBottomIndex)
+            _updateChildren = true;
+
+        // When scrolling only rebuild visible list when a new item should be visible
+        if (_updateChildren)
+        {
+            _updateChildren = false;
+
+            var toRemove = new Dictionary<ListData, ListContainerButton>(_buttons);
+            foreach (var child in Children.ToArray())
+            {
+                if (child == _vScrollBar)
+                    continue;
+                RemoveChild(child);
+            }
+
+            if (_data.Count > 0)
+            {
+                for (var i = _topIndex; i < _bottomIndex; i++)
+                {
+                    var data = _data[i];
+
+                    if (_buttons.TryGetValue(data, out var button))
+                        toRemove.Remove(data);
+                    else
```

Comment:
- toRemove is always an empty dictionary, and afaict can just be removed.
- _updateChildren is also used when PopulateList is called, so it should always update all children to ensure that items are re-generated. Or populate list should just clear _data or something like that.
- Currently this causes an issue where the bwoing menu player names strings / unread-symbols don't get updated because `GenerateItem -> GenerateButton -> OverrideText` never get called